src/sentry/issues/issue_occurrence.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+import hashlib
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence
+
+from dateutil.parser import parse as parse_date
+from django.utils.timezone import is_aware
+
+from sentry import nodestore
+from sentry.types.issues import GroupType
+
+
+@dataclass(frozen=True)
+class IssueEvidence:
+    name: str
+    value: str
+    # Whether to prioritise displaying this evidence to users over other issue evidence. Should
+    # only be one important row per occurrence.
+    important: bool
+
+    def to_dict(
+        self,
+    ) -> Mapping[str, str | bool]:
+        return {
+            "name": self.name,
+            "value": self.value,
+            "important": self.important,
+        }
+
+
+@dataclass(frozen=True)
+class IssueOccurrence:
+    """
+    A class representing a specific occurrence of an issue. Separate to an `Event`. An `Event` may
+    have 0-M `IssueOccurrences` associated with it, and each `IssueOccurrence` is associated with
+    one `Event`.
+
+    Longer term, we might change this relationship so that each `IssueOccurrence` is the primary
+    piece of data that is passed around. It would have an `Event` associated with it.
+    """
+
+    id: str
+    # Event id pointing to an event in nodestore
+    event_id: str
+    fingerprint: Sequence[str]
+    issue_title: str
+    # Exact format not decided yet, but this will be a string regardless
+    subtitle: str
+    resource_id: str | None
+    # Extra context around how the problem was detected. Used to display grouping information on
+    # the issue details page, and will be available for use in UI customizations.
+    evidence_data: Mapping[str, Any]
+    # Extra context around the problem that will be displayed as a default in the UI and alerts.
+    # This should be human-readable. One of these entries should be marked as `important` for use
+    # in more space restricted integrations.
+    evidence_display: Sequence[IssueEvidence]
```

Comment:
do we have a length cap on this? max 10 pieces of evidence or something?

Reply:
I think we probably have some kind of cap here, but we'll handle it as part of ingestion. Similarly we probably want to limit the length of values within each piece of evidence as well

Reply:
right yeah

Reply:
Does this need to be `Sequence[IssueEvidence | None]` or however you type that? Other than that I think I'm tapped on comments :D

Reply:
Typing doesn't care how many elements there are in the list, so `Sequence[IssueEvidence]` is ok here. If we had `Sequence[IssueEvidence | None]` then entries in the list could be `None`, which we don't want